### PR TITLE
Ansible Vault and Azure Key Vault vault password script

### PR DESCRIPTION
**Pull request by OxHobbs**
##### SUMMARY
Adding a new contrib script that provides the capability to use an Azure KeyVault secret with the Ansible Vault.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_vault vault password file script

##### ANSIBLE VERSION
```
ansible 2.5.4
  config file = /Users/ox/.ansible.cfg
  configured module search path = [u'/Users/ox/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.10 (default, Oct  6 2017, 22:29:07) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### ADDITIONAL INFORMATION
This script may be used as the password file provided to the `ansible-vault` command.  This allows the user to store the vault password securely in Azure Key Vault.

Made heavy re-use of the AzureRM class from the Azure Dynamic Inventory script written by @chouseknecht and @nitzmahone for authentication and configuration style.

```
ansible-vault create django-secrets.yml --vault-password-file ./azure_vault.py
```

### CODE REVIEW

**Comment by francois-travais** [comment]
Does someone have an idea when this PR will be merged?

File: contrib/vault/azure_vault.py [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+            self.fail("Failed to get credentials. Either pass as parameters, set environment variables, "
+                      "or define a profile in ~/.azure/credentials.")
+
+        # if cloud_environment specified, look up/build Cloud object
+        raw_cloud_env = self.credentials.get('cloud_environment')
+        if not raw_cloud_env:
+            self._cloud_environment = azure_cloud.AZURE_PUBLIC_CLOUD  # SDK default
+        else:
+            # try to look up "well-known" values via the name attribute on azure_cloud members
+            all_clouds = [x[1] for x in inspect.getmembers(azure_cloud) if isinstance(x[1], azure_cloud.Cloud)]
```
Comment (maxamillion) [inline contrib/vault/azure_vault.py:220]: Will the azure API always return a non-zero list or could this result in an `IndexError`? (If so, please wrap in `try`/`except`)

Comment (OxHobbs) [reply contrib/vault/azure_vault.py:220]: The way I understand it, these are hard coded in the msrestazure package, so as long as the package imports it should always be non-zero.

**Comment by maxamillion** [comment]
rebuild_merge

**Review by maxamillion — APPROVED** [review]

**Pull request merged**